File: mobile_app/lib/features/vehicle/models/vehicle.dart
class Vehicle {
  final int? id;
  final String brand;
  final String vehicleType;
  final String vehicleName;

  // 3 New data fields
  final String licensePlate;
  final int? engineCapacity;
  final int? currentKm;

  final int? ownerId;

  const Vehicle({
    this.id,
    required this.brand,
    required this.vehicleType,
    required this.vehicleName,
    required this.licensePlate,
    this.engineCapacity,
    this.currentKm,
    this.ownerId,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) => Vehicle(
    id: json['id'] as int?,
    brand: json['brand'] as String? ?? '',
    vehicleType: json['vehicleType'] as String? ?? 'XE_TAY_GA',
    vehicleName: json['vehicleName'] as String? ?? '',
    licensePlate: json['licensePlate'] as String? ?? '',
    engineCapacity: json['engineCapacity'] as int?,
    currentKm: json['currentKm'] as int?,
    ownerId: (json['owner'] as Map<String, dynamic>?)?['id'] as int?,
  );

  Map<String, dynamic> toJson() => {
    'brand': brand,
    'vehicleType': vehicleType,
    'vehicleName': vehicleName,
    'licensePlate': licensePlate,
    'engineCapacity': engineCapacity ?? 0,
    'currentKm': currentKm ?? 0,
  };

  String get typeLabel => vehicleType == 'XE_SO' ? 'Manual' : 'Scooter';
}

File: backend-java/src/main/java/com/carebike/backend/features/vehicle/entity/Vehicle.java
package com.carebike.backend.features.vehicle.entity;

import com.carebike.backend.features.auth.entity.User;
import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "vehicles")
@Data
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    /** Hãng xe — e.g., Honda, Yamaha, Suzuki, SYM */
    @Column(nullable = false, columnDefinition = "NVARCHAR(100)")
    private String brand;

    /** Dòng xe — XE_SO or XE_TAY_GA */
    @Column(name = "vehicle_type", nullable = false, length = 50)
    private String vehicleType;

    /** Tên xe — e.g., Airblade, Exciter, Raider */
    @Column(name = "vehicle_name", nullable = false, columnDefinition = "NVARCHAR(150)")
    private String vehicleName;
    
    /** Biển số xe */
    @Column(name = "license_plate", length = 20)
    private String licensePlate;

    /** Phân khối (cc) */
    @Column(name = "engine_capacity")
    private Integer engineCapacity;

    /** Số km đã đi */
    @Column(name = "current_km")
    private Integer currentKm;

    @ManyToOne
    @JoinColumn(name = "owner_id", referencedColumnName = "id", nullable = false)
    private User owner;
}